hoki/utils/binning.py

```diff
@@ -0,0 +1,180 @@
+"""
+Routines to bin spectra.
+
+Author: Martin Glatzle
+"""
+import numpy as np
+from numba import jit
+
+
+def bin_spectra(wave, spectra, bins, edges=False):
+    """
+    Bin spectra conserving luminosity.
+
+    Given SEDs sampled at certain wavelengths/frequencies will compute their
+    values in given wavelength/frequency bins. The new SED values are bin
+    averages computed using trapezoidal integration. This ensures that the
+    luminosity per bin is conserved. Of course, only downsampling really makes
+    sense here, i.e. the input SEDs should be well sampled compared to the
+    desired output bins.
+
+    Effectively converts input spectra to step functions of
+    wavelength/frequency. Note, in particular, that this means that only
+    rectangle rule integration can sensibly be performed on the output
+    spectra. Higher order integration methods are not meaningful.
+
+    Parameters
+    ----------
+    wave : `numpy.ndarray` (N_wave,)
+        Wavelengths or frequencies at which spectra are known.
+    spectra : `numpy.ndarray` (N, N_wave)
+        The SEDs to resample given as L_lambda [Energy/Time/Wavelength] or L_nu
+        [Energy/Time/Frequency] in accordance with `wave`.
+    bins : `numpy.ndarray` (N_bins,)
```

Comment:
A little clarification needed in this definition. 

Just reading this it's not clear whether i can give the number of *old bins* I want in *new bins* (e.g. combining 3 bins into one) or if I have to give central wavelength (or edges, that latter part is clear).  If the option is central wavelength or edge wavelengths I suggest a minor refactor:

instead of having a `bins` and `edges`  parameter, have a `wl_center` and `wl_edges` parameters - only one should be filled and you an check for that. Default them to None (`wl_center=None` and `wl_edges=None`) and if both remain non on execution print a custom error message with a 'HOKI DEBUGGER:" explaining to the use they need to provide at least one of these.

Reply:
Fair point. I've overhauled how the bins work, bringing it more in line with what one knows from `numpy` and also updated the docstrings. Should hopefully be much clearer now.